src/ec2_metadata/__init__.py

```diff
@@ -97,6 +97,19 @@ def iam_info(self) -> Optional[Dict[str, Any]]:
             return None
         return resp.json()
 
+    @cached_property
+    def iam_security_credentials(self) -> Optional[Dict[str, Any]]:
+        instance_profile_name = self.instance_profile_name
+        if instance_profile_name is None:
+            return None
+        resp = self._get_url(
+            METADATA_URL + "iam/security-credentials/" + instance_profile_name,
+            allow_404=True,
+        )
+        if resp.status_code == 404:
+            return None
+        return resp.json()
```

Comment:
is it really useful to return an unstructured dict from [the JSON](
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials)? Wouldn't it be better to return at least a structured object?